### PR TITLE
dx | 1169 getter methods implementation

### PR DESCRIPTION
**Pull request by harshithad0703**
- implemented getter methods for developer hub, launch and personalize
- replaced getDeveloperHubUrl method with getter method in export and import plugin
- replaced getLaunchHubUrl method with getter method in launch plugin
- @contentstack/cli-cm-export: 1.11.6 -> 1.12.0
- @contentstack/cli-cm-import: 1.16.5 -> 1.17.0
- @contentstack/cli-command: 1.2.19 -> 1.3.0
- @contentstack/cli-launch: 1.2.0 -> 1.3.0

### CODE REVIEW

File: packages/contentstack-export/src/export/modules-js/marketplace-apps.js [hunk 1]
```diff
@@ -43,7 +45,7 @@ module.exports = class ExportMarketplaceApps {
       return Promise.resolve();
     }
 
-    this.developerHubBaseUrl = this.config.developerHubBaseUrl || (await getDeveloperHubUrl(this.config));
+    this.developerHubBaseUrl = this.config.developerHubBaseUrl || (command.developerHubUrl);
```
Comment (shafeeqd959) [inline packages/contentstack-export/src/export/modules-js/marketplace-apps.js:48]: @harshithad0703 @aman19K do we need to update in the old js

Comment (shafeeqd959) [reply packages/contentstack-export/src/export/modules-js/marketplace-apps.js:48]: @aman19K is this backward compatible, for the existing users >?

File: packages/contentstack-export/src/export/modules/marketplace-apps.ts [hunk 2]
```diff
@@ -47,7 +49,7 @@ export default class ExportMarketplaceApps {
       this.marketplaceAppConfig.dirName,
     );
     await fsUtil.makeDirectory(this.marketplaceAppPath);
-    this.developerHubBaseUrl = this.exportConfig.developerHubBaseUrl || (await getDeveloperHubUrl(this.exportConfig));
+    this.developerHubBaseUrl = this.exportConfig.developerHubBaseUrl || this.command.developerHubUrl;
```
Comment (shafeeqd959) [inline packages/contentstack-export/src/export/modules/marketplace-apps.ts:52]: Creating a command instance to access the developerhub url is not correct way here, we can get it from the parent class

File: packages/contentstack-import/src/import/modules/marketplace-apps.ts [hunk 3]
```diff
@@ -94,7 +95,7 @@ export default class ImportMarketplaceApps {
       return Promise.resolve();
     }
     await fsUtil.makeDirectory(this.mapperDirPath);
-    this.developerHubBaseUrl = this.importConfig.developerHubBaseUrl || (await getDeveloperHubUrl(this.importConfig));
+    this.developerHubBaseUrl = this.importConfig.developerHubBaseUrl || this.command.developerHubUrl;
```
Comment (shafeeqd959) [inline packages/contentstack-import/src/import/modules/marketplace-apps.ts:98]: Same as above

File: packages/contentstack-launch/src/base-command.ts [hunk 4]
```diff
@@ -105,7 +106,7 @@ export abstract class BaseCommand<T extends typeof Command> extends Command {
       this.flags['data-dir'] || this.flags.config
         ? this.flags.config || resolve(this.flags['data-dir'], config.configName)
         : resolve(process.cwd(), config.configName);
-    const baseUrl = config.launchBaseUrl || getLaunchHubUrl();
+    const baseUrl = config.launchBaseUrl || this.command.launchHubUrl;
```
Comment (shafeeqd959) [inline packages/contentstack-launch/src/base-command.ts:109]: Same as above

File: packages/contentstack-export/src/export/modules-js/marketplace-apps.js (hunk 1 above)
Comment (aman19K) [reply packages/contentstack-export/src/export/modules-js/marketplace-apps.js:48]: @shafeeqd959 don't think we need to made changes in js code.

**Comment by aman19K** [comment]
@harshithad0703 , since we're only maintaining TS One, there's no need to modify the export and import JS code.
CC: @shafeeqd959

File: packages/contentstack-export/src/export/modules-js/marketplace-apps.js (hunk 1 above)
Comment (harshithad0703) [reply packages/contentstack-export/src/export/modules-js/marketplace-apps.js:48]: removed js changes

File: packages/contentstack-import/src/import/modules/marketplace-apps.ts (hunk 3 above)
Comment (harshithad0703) [reply packages/contentstack-import/src/import/modules/marketplace-apps.ts:98]: done

File: packages/contentstack-launch/src/base-command.ts (hunk 4 above)
Comment (harshithad0703) [reply packages/contentstack-launch/src/base-command.ts:109]: done

File: packages/contentstack-export/src/export/modules/marketplace-apps.ts (hunk 2 above)
Comment (harshithad0703) [reply packages/contentstack-export/src/export/modules/marketplace-apps.ts:52]: yes got it

File: packages/contentstack-launch/src/base-command.ts [hunk 5]
```diff
@@ -105,7 +105,7 @@ export abstract class BaseCommand<T extends typeof Command> extends Command {
       this.flags['data-dir'] || this.flags.config
         ? this.flags.config || resolve(this.flags['data-dir'], config.configName)
         : resolve(process.cwd(), config.configName);
-    const baseUrl = config.launchBaseUrl || getLaunchHubUrl();
+    const baseUrl = config.launchBaseUrl || this.launchHubUrl;
```
Comment (aman19K) [inline packages/contentstack-launch/src/base-command.ts:108]: @harshithad0703 , what if baseUrl is undefined ? pls handle this condition. like below 
let baseUrl = config.launchBaseUrl || this.launchHubUrl;

if (!baseUrl) {
    baseUrl = getLaunchHubUrl();
}

**Review by aman19K — DISMISSED** [review]

File: packages/contentstack-launch/src/base-command.ts (hunk 5 above)
Comment (harshithad0703) [reply packages/contentstack-launch/src/base-command.ts]: done

**Review by aman19K — APPROVED** [review]

**Pull request merged**